### PR TITLE
MGMT-7947: Adding Openshift template

### PR DESCRIPTION
**Pull request by michaellevy101**
Adding Openshift template will be used by the app-sre to deploy this service on the different environments

### CODE REVIEW

File: openshift/template.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          - name: ES_SERVER
+              valueFrom:
+                secretKeyRef:
+                  key: endpoint
+                  name: assisted-installer-elasticsearch
+          - name: ES_INDEX
+            value: ${ES_INDEX}
+parameters:
+- name: IMAGE_NAME
+  value: quay.io/app-sre/
```
Comment (omertuc) [inline openshift/template.yaml:52]: `image: ${IMAGE_NAME}:${IMAGE_TAG}`
==
`quay.io/app-sre/:`

doesn't seem right

File: events_scrap/events_scrap.py [hunk 2]
```diff
@@ -37,7 +39,11 @@ def __init__(self, inventory_url: str, offline_token: str, index: str, es_server
         self.client = ClientFactory.create_client(url=self.inventory_url, offline_token=offline_token)
 
         self.index = index
-        self.es = elasticsearch.Elasticsearch(es_server, http_auth=(es_user, es_pass))
+
+        if es_user:
```
Comment (osherdp) [inline events_scrap/events_scrap.py:42]:
```suggestion
        http_auth = None
        if es_user:
```
to make sure we don't have any undefined variable

File: README.md [hunk 3]
```diff
@@ -12,11 +12,9 @@ It pulls all the cluster events and merge it with the existing data.
...
-| `ES_PASS`          | Elasticsearch password  |  |
+| `ES_USER(optional)`          | Elasticsearch user name | elastic |
+| `ES_PASS(optional)`          | Elasticsearch password  |  |
 | `REMOTE_SERVICE_URL` | Assisted service url  | https://api.openshift.com |
 | `OFFLINE_TOKEN` | Assisted service offline token  | |
-
-
-
-
+| `INVENTORY_URL` | Assisted service inventory url  | |
```
Comment (osherdp) [inline README.md:19]: can you put an example in the table?

File: README.md [hunk 4]
```diff
@@ -12,11 +12,9 @@ It pulls all the cluster events and merge it with the existing data.
...
+| `BACK_DESTINATION` | Path to save backup, if empty no back up saved  | |
```
Comment (osherdp) [inline README.md:20]:
```suggestion
| `BACKUP_DESTINATION` | Path to save backup, if empty no backups will be saved  | |
```

File: events_scrap/events_scrap.py [hunk 5]
```diff
@@ -207,31 +213,29 @@ def process_event_doc(event_data, cluster_bash_data):
...
-
-    return parser.parse_args()
+    return {
+        "inventory_url": utils.get_env("INVENTORY_URL", False),
+        "offline_token": utils.get_env("OFFLINE_TOKEN", True),
+        "es_server": utils.get_env("ES_SERVER", True),
+        "es_user": utils.get_env("ES_USER", False),
+        "es_pass": utils.get_env("ES_PASS", False),
+        "index": utils.get_env("ES_INDEX", True),
+        "backup-destination": utils.get_env("BACK_DESTINATION", False),
```
Comment (osherdp) [inline events_scrap/events_scrap.py:222]:
```suggestion
        "backup_destination": utils.get_env("BACKUP_DESTINATION", False),
```

File: openshift/template.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+- name: IMAGE_TAG
+  value: ''
+  required: true
+- name: REPLICAS_COUNT
+  value: "1"
+- name: ES_INDEX
+  value: ''
```
Comment (osherdp) [inline openshift/template.yaml:59]: nit: can you change your IDE settings to end with blank lines?

File: openshift/template.yaml (hunk 1 above)
Comment (michaellevy101) [reply openshift/template.yaml:52]: nice catch, thanks:)

File: openshift/template.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: assisted-events-scrape
```
Comment (osherdp) [inline openshift/template.yaml:4]: any chance we can change the repo name?

File: openshift/template.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+objects:
```
Comment (osherdp) [inline openshift/template.yaml:5]: any chance we can add ``pdb`` as well?
to make sure there's always one instance up during cluster upgrades

**Review by osherdp — CHANGES_REQUESTED** [review]
looking good, just a few comments

File: README.md (hunk 3 above)
Comment (michaellevy101) [reply README.md:19]: not my code, just documented it, don't have an example yet. will do after running it

File: openshift/template.yaml (hunk 8 above)
Comment (michaellevy101) [reply openshift/template.yaml:5]: sure, I'm focusing on deploy it first ASAP, please add a new task for me. 
https://issues.redhat.com/browse/MGMT-7096

File: openshift/template.yaml (hunk 7 above)
Comment (michaellevy101) [reply openshift/template.yaml:4]: changed the repo name, and update openshift/release as well
https://github.com/openshift/release/pull/23441

Comment (osherdp) [reply openshift/template.yaml:4]: Thank you!
Let me know when you need review for the change in gitlab

Comment (michaellevy101) [reply openshift/template.yaml:4]: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/28417

File: openshift/template.yaml (hunk 8 above)
Comment (michaellevy101) [reply openshift/template.yaml:5]: done

File: events_scrap/events_scrap.py (hunk 5 above)
Comment (osherdp) [reply events_scrap/events_scrap.py]: I think you still didn't handle that
in another place you referred to it as ``backup_destination``

**Review by osherdp — CHANGES_REQUESTED** [review]
just one last comment

**Review by osherdp — APPROVED** [review]

**Pull request merged**